### PR TITLE
Implement homogeneous xml formatting for NIRSpec, NIRISS, FGS.

### PR DESCRIPTION
**Pull request by Johannes-Sahlmann**
Add function to extract field formats.

### CODE REVIEW

**Comment by Johannes-Sahlmann** [comment]
@shanosborne This is ready to merge, since all associated Jira tickets were approved. This will place the updated SIAF xml files in the pre-delivery-data folder from where we can grab them for the PRD Change Request.

File: generate/homogenize_xml_field_format.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+            print('\nRunning nirspec_aperture_transforms test for pre_delivery_siaf')
+            test_nirspec.test_nirspec_aperture_transforms(siaf=pre_delivery_siaf, verbose=False)
+
+            print('\nRunning nirspec_slit_transforms test for pre_delivery_siaf')
+            test_nirspec.test_nirspec_slit_transformations(siaf=pre_delivery_siaf, verbose=False)
+
+show_field_formats = True
+
+if show_field_formats:
```
Comment (shanosborne) [inline generate/homogenize_xml_field_format.py]: Could this also be a separate function that can get called in the file? Just to keep things separate

File: generate/homogenize_xml_field_format.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    reference_instrument_name = 'NIRISS'
+
+    reference_table = xml_formats[reference_instrument_name]
+    # 1/0
```
Comment (shanosborne) [inline generate/homogenize_xml_field_format.py]: remove commented out code

File: generate/homogenize_xml_field_format.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    tag_list = list(reference_table['field_name'])
+    rows_to_delete = [i for i in range(len(reference_table)) if ('Sci2IdlX' in tag_list[i] or
+                                                                 'Sci2IdlY' in tag_list[i] or
+                                                                 'Idl2SciX' in tag_list[i] or
+                                                                 'Idl2SciY' in tag_list[i]) and (tag_list[i] not in ['Sci2IdlX00', 'Idl2SciY00'])]
+
+    reference_table.remove_rows(rows_to_delete)
+
+    reference_table[columns_to_show].pprint()
+    # reference_table[columns_to_show].write('siaf_field_formats.txt', format='ascii.fixed_width',
```
Comment (shanosborne) [inline generate/homogenize_xml_field_format.py]: remove commented out code?

File: generate/homogenize_xml_field_format.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    tree = ET.parse(filename)
+
+    # generate Aperture objects from SIAF XML file, parse the XML
+    for entry in tree.getroot().iter('SiafEntry'):
+        show = False
+        field_index = 1
+        for node in entry.iterchildren():
+            # show = False
+            field_number = '1.{:d}'.format(field_index)
+            # print('{}   {}: {}'.format(primary_master_aperture_name, node.tag, node.text))
```
Comment (shanosborne) [inline generate/homogenize_xml_field_format.py]: Remove commented out code

File: generate/homogenize_xml_field_format.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    Johannes Sahlmann
+
+"""
+import os
+
+import pysiaf
+from pysiaf.utils import polynomial, tools, compare
+from pysiaf.constants import JWST_SOURCE_DATA_ROOT, JWST_TEMPORARY_DATA_ROOT, JWST_DELIVERY_DATA_ROOT
+# from pysiaf import iando
```
Comment (shanosborne) [inline generate/homogenize_xml_field_format.py]: Remove commented out code

File: generate/homogenize_xml_field_format.py (hunk 1 above)
Comment (Johannes-Sahlmann) [reply generate/homogenize_xml_field_format.py]: done

File: generate/homogenize_xml_field_format.py (hunk 3 above)
Comment (Johannes-Sahlmann) [reply generate/homogenize_xml_field_format.py]: done

**Pull request merged**